addons/spectator/functions/fnc_camera.sqf

```diff
@@ -0,0 +1,1081 @@
+/*
+    Author:
+    voiper, derived in part from BIS_fnc_camera by Karel Moricky
+
+    Description:
+    Spectator camera and UI.
+
+    Arguments:
+    0: Mode: "Init" is the only mission relevant one <String>
+    1: <Array> (optional):
+        0: Whether player can escape from camera (true for MP spectator; false for SP photography) <Bool>
+
+    Example:
+    ["Init", [false]] call ace_spectator_fnc_camera;
+
+    Return Value:
+    None
+
+    Public:
+    No
+*/
+
+#include "script_component.hpp"
+#include "\a3\editor_f\Data\Scripts\dikCodes.h"
+
+disableSerialization;
+_mode = _this select 0;
+_this = _this select 1;
+
+switch _mode do {
+
+    case "Init": {
+    
+        GVAR(noEscape) = if (count _this > 0) then {_this select 0} else {false};
+        
+        call FUNC(penPos);
+
+        _camPos = if (!isNil QGVAR(startingPos)) then {
+            GVAR(startingPos)
+        } else {
+            getPos cameraOn
+        };
+        
+        _camDir = if (!isNil QGVAR(startingDir)) then {
+            GVAR(startingDir)
+        } else {
+            0
+        };
+        
+        _camPos set [2, (_camPos select 2) + 2];
+        _cam = "camera" camCreate _camPos;
+        _cam setDir _camDir;
+        _cam cameraEffect ["internal", "back"];
+        _cam camSetFocus [-1, -1];
+        _cam camCommit 0;
+        showCinemaBorder false;
+        cameraEffectEnableHUD true;
+        setViewDistance 3000;
```

Comment:
Yeah I'd suggest that it's disabled by default and let the mission makers allow it through a module option for those cases where changing VD is 'okay'